File: src/helio.pl
#! /usr/bin/perl

use strict;
use warnings;

use Astro::FITS::CFITSIO;
use Data::Dumper;

use Astrolib 'helio_jd';
use Chandra::Tools::Common 'read_bintbl_cols';

@ARGV == 1 or die "Usage: $0 evtfile\n";
my $evt = shift;

my $hdr = Astro::FITS::CFITSIO::fits_read_header($evt.'[events]');

my $ra = $hdr->{'RA_TARG'} + 0.;
my $dec = $hdr->{'DEC_TARG'} + 0.;

my ($time) = read_bintbl_cols($evt, 'time', { extname => 'events' });
my $mjd = $time/86400 + $hdr->{'MJDREF'};
my $mjd_helio = helio_jd($mjd+0.5, $ra, $dec) - 0.5;
print+($mjd_helio-$mjd)->slice('0:3'),"\n";

File: src/helio.py
import argparse
import astropy.io.fits
from astropy import time, coordinates as coord, units as u
import numpy as np

def read_evt(evt):
    with astropy.io.fits.open(evt) as hdulist:
        hdu = hdulist['events']
        data = hdu.data
        hdr = hdu.header
        ra = hdr['ra_targ']
        dec = hdr['dec_targ']
        mjd = hdr['mjdref'] + data['time']/86400
        return ra, dec, mjd

def helio(args):
    ra, dec, mjd = read_evt(args.evt)
    ip_peg = coord.SkyCoord(ra, dec, unit=u.deg)
    greenwich = coord.EarthLocation.of_site('greenwich')
    times = time.Time(mjd, format='mjd')
    ltt_helio = times.light_travel_time(ip_peg, 'heliocentric',
                                        location=greenwich)
    print(ltt_helio)

def main():
    parser = argparse.ArgumentParser(
        description='Convert event times to heliocentric JD',
    )
    parser.add_argument('evt', help='EVT file.')
    args = parser.parse_args()
    helio(args)

if __name__ == '__main__':
    main()
